### PR TITLE
Prevent IDE0290 (convert to primary constructor) in ref struct

### LINKED ISSUE #73614: IDE0290 code fix is invalid in `ref struct`
**Version Used**: '4.11.0-1.24229.8 (28985c0a)'. Language version: 12.0

**Steps to Reproduce**:

https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoTAnApgMwAIBnAF2wFcBjUwgZQjAAcAbXTAb00J8KeygA3CKVyEASrggATAPIwWATzpMIMADxUAFhGwA+QgH0y2ANyZuvBAGZ6jVrgAUkmfKUq1mnfpLkAlJY8XBi8oUYmhAC8vmaBhAC+FiFWtggALIQAsgD2grgAcrgAHqSOAclBcaHC2CSqMFHh5OYVCZiJGEA

````c#
ref struct Sample
{
    private ReadOnlySpan<char> _str;

    public Sample(ReadOnlySpan<char> str)
    {
        _str = str;
    }

    public void MoveNext()
    {
        var span = _str;
    }
}
````

Use the fix to convert to primary constructor and remove field

````c#
ref struct Sample(ReadOnlySpan<char> str)
{
    public void MoveNext()
    {
        // ❌ CS9110 Cannot use primary constructor parameter 'str' that has ref-like type inside an instance member
        var span = str;
    }
}
````

**Diagnostic Id**: IDE0290 Use primary Constructor

**Expected Behavior**: The code fix is not suggested

**Actual Behavior**: The code fix is suggested

### PR DESCRIPTION
**Pull request by GiovanniBraconi**
Fixes #73614

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Hi, could you not force-push :)  It breaks the code flow process.  THanks!  

Feel free to squash merge at the end if you want though.

**Review by CyrusNajmabadi — APPROVED** [review]
Preliminary sign-off, with request to just change the comment.  Thanks!

File: src/Analyzers/CSharp/Analyzers/UsePrimaryConstructor/CSharpUsePrimaryConstructorDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -366,6 +366,10 @@ void RegisterFieldOrPropertyAnalysisIfNecessary(Analyzer? analyzer)
                         if (primaryConstructor != null)
                             return false;
 
+                        // Check if any of the parameters is a ref-like type
```
[the remark this answers is not in the document]
Comment (GiovanniBraconi) [reply src/Analyzers/CSharp/Analyzers/UsePrimaryConstructor/CSharpUsePrimaryConstructorDiagnosticAnalyzer.cs]: You're right; i deleted it 😄

**Comment by CyrusNajmabadi** [comment]
AWesome.  Thanks so much for the contribution.  Hope to work with you more in the future! :)

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**